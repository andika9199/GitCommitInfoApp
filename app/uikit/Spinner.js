import React from 'react';
import {ActivityIndicator, Platform} from 'react-native';
import PropTypes from 'prop-types';
import {BLACK} from '../configs/style';

/**
 *
 * @param {boolean} animating
 * @param {string} color
 * @param {string|number} size
 * @param {number|Object|array} style
 */

function Spinner(props) {
  const {animating, color, style, size} = props;
  const chooseSize = size || 'large';
  const renderedSize = Platform.OS === 'ios' ? chooseSize : size;

  return (
    <ActivityIndicator
      animating={animating}
      color={color}
      size={renderedSize}
      style={style}
    />
  );
}

Spinner.propTypes = {
  animating: PropTypes.bool,
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.string]),
  style: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.object,
    PropTypes.array,
  ]),
};

Spinner.defaultProps = {
  animating: true,
  color: BLACK,
  size: 'large',
};

export default Spinner;
