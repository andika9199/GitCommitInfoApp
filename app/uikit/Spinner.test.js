import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from './Spinner';

it('Spinner successfully rendered', () => {
  const component = renderer.create(<Spinner />);
  expect(component).toMatchSnapshot();
});
