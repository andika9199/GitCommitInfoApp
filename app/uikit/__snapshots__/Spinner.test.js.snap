// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner successfully rendered 1`] = `
<ActivityIndicator
  animating={true}
  color="#333333"
  hidesWhenStopped={true}
  size="large"
/>
`;
